discord/state.py

```diff
@@ -1311,32 +1353,32 @@ def get_channel(self, id):
             if channel is not None:
                 return channel
 
-    def create_message(self, *, channel, data):
+    def create_message(self, *, channel: Union[TextChannel, Thread, DMChannel, GroupChannel, PartialMessageable], data: MessagePayload) -> Message:
         return Message(state=self, channel=channel, data=data)
 
 class AutoShardedConnectionState(ConnectionState):
-    def __init__(self, *args, **kwargs):
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
         super().__init__(*args, **kwargs)
-        self._ready_task = None
-        self.shard_ids = ()
-        self.shards_launched = asyncio.Event()
+        self.shard_ids: Union[List[int], range] = []
+        self.shards_launched: asyncio.Event = asyncio.Event()
 
-    def _update_message_references(self):
-        for msg in self._messages:
+    def _update_message_references(self) -> None:
+        for msg in self._messages or []:
```

Comment:
When this call happens `self._messages` is never `None` since it's in an `if` block.